source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -0,0 +1,331 @@
+//
+// TM & (c) 2021 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXGenShader/OCIOColorManagementSystem.h>
+
+#include <MaterialXGenShader/GenContext.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+
+#include <iostream>
+
+#ifdef OpenColorIO_SKIP_IMPORTS
+#error "OCIO has been build as a non-static library"
+#endif
+#pragma warning(disable: 4275)
+#include <OpenColorIO/OpenColorIO.h>
+namespace OCIO = OCIO_NAMESPACE;
+
+namespace MaterialX
+{
+
+const string OCIOColorManagementSystem::CMS_NAME = "ocio_cms";
+
+// Defines an unsupported language.
+const int OCIO_UNSUPPORT_LANGUAGE = -1;
+
+// Mapping from supported MaterialX targets to OCIO shading languages
+using OCIOLanguageMap = std::unordered_map<std::string, OCIO::GpuLanguage>;
+
+// OCIO related information
+class OCIOInformation
+{
+  public:
+    // OCIO config file to use
+    OCIO::ConstConfigRcPtr config = nullptr;
+    // Target to language map
+    OCIOLanguageMap languageMap;
+    // Language used for code generation (is a OCIO::GpuLanguage)
+    int language = OCIO_UNSUPPORT_LANGUAGE;
+    // MaterialX target
+    std::string target;
+};
+
+OCIOColorManagementSystemPtr OCIOColorManagementSystem::create(const string& target)
+{
+    return OCIOColorManagementSystemPtr(new OCIOColorManagementSystem(target));
+}
+
+OCIOColorManagementSystem::OCIOColorManagementSystem(const string& target) :
+    _target(target)
+{
+    _ocioInfo = new OCIOInformation();
+    _ocioInfo->language = OCIO_UNSUPPORT_LANGUAGE;
+    _ocioInfo->config = nullptr;
+    _ocioInfo->target = EMPTY_STRING;
+    _ocioInfo->languageMap["genglsl"] = OCIO::GPU_LANGUAGE_GLSL_4_0;
+}
+
+OCIOColorManagementSystem::~OCIOColorManagementSystem()
+{
+    delete _ocioInfo;
+}
+
+bool OCIOColorManagementSystem::isValid() const
+{
+    return (_document && _ocioInfo->config && _ocioInfo->language != OCIO_UNSUPPORT_LANGUAGE);
+}
+
+void OCIOColorManagementSystem::loadLibrary(DocumentPtr document) 
+{
+    _ocioInfo->language = OCIO_UNSUPPORT_LANGUAGE;
+    _ocioInfo->target = EMPTY_STRING;
+
+    _document = document;
+    if (!_document)
+    {
+        throw ExceptionShaderGenError("No library loaded for color management system");
+    }
+
+    TargetDefPtr targetDef = _document->getTargetDef(_target);
+    const StringVec targets = targetDef->getMatchingTargets();
```

Comment:
Could add a nullptr check on targetdDef, to fail with an exception instead of crashing if a targetdef for the target has not been loaded.

Reply:
Yes. I'll fix this.